File: backend/app/models/case.py
from sqlalchemy import Column, BigInteger, String, DateTime, ForeignKey, Integer, Numeric
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class Case(Base):
    """案件主表"""
    __tablename__ = "cases"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    case_code = Column(String(100), unique=True, nullable=False, index=True, comment='案件唯一标识')
    tenant_id = Column(BigInteger, ForeignKey('tenants.id'), nullable=False, index=True, comment='所属甲方ID')
    agency_id = Column(BigInteger, ForeignKey('collection_agencies.id'), index=True, comment='所属催收机构ID')
    team_id = Column(BigInteger, ForeignKey('collection_teams.id'), index=True, comment='所属催收小组ID')
    collector_id = Column(BigInteger, ForeignKey('collectors.id'), index=True, comment='分配催员ID')
    queue_id = Column(BigInteger, ForeignKey('case_queues.id'), index=True, comment='所属队列ID')
    user_id = Column(String(100), index=True, comment='用户编号')
    user_name = Column(String(100), comment='用户姓名')
    mobile = Column(String(50), comment='手机号')
    case_status = Column(String(50), index=True, comment='案件状态：pending_repayment/partial_repayment/normal_settlement/extension_settlement')
    overdue_days = Column(Integer, index=True, comment='逾期天数（用于自动分配队列）')
    loan_amount = Column(Numeric(15, 2), comment='贷款金额')
    repaid_amount = Column(Numeric(15, 2), default=0, comment='已还款金额')
    outstanding_amount = Column(Numeric(15, 2), comment='逾期金额')
    due_date = Column(DateTime, index=True, comment='到期日期')
    settlement_date = Column(DateTime, comment='结清日期')
    assigned_at = Column(DateTime, index=True, comment='分配时间')
    last_contact_at = Column(DateTime, comment='最后联系时间')
    next_follow_up_at = Column(DateTime, comment='下次跟进时间')
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # 关系
    tenant = relationship("Tenant", back_populates="cases")
    agency = relationship("CollectionAgency", back_populates="cases")
    team = relationship("CollectionTeam", back_populates="cases")
    collector = relationship("Collector", back_populates="cases")
    queue = relationship("CaseQueue", back_populates="cases")
    standard_field_values = relationship("CaseStandardFieldValue", back_populates="case", cascade="all, delete-orphan")
    custom_field_values = relationship("CaseCustomFieldValue", back_populates="case", cascade="all, delete-orphan")
    assignment_history = relationship("CaseAssignmentHistory", back_populates="case", cascade="all, delete-orphan")
    # 新增关系
    communication_records = relationship("CommunicationRecord", back_populates="case", cascade="all, delete-orphan")
    ptp_records = relationship("PTPRecord", back_populates="case", cascade="all, delete-orphan")
    quality_inspections = relationship("QualityInspectionRecord", back_populates="case", cascade="all, delete-orphan")
    contacts = relationship("CaseContact", back_populates="case", cascade="all, delete-orphan")


class CaseStandardFieldValue(Base):
    """标准字段值表"""
    __tablename__ = "case_standard_field_values"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    case_id = Column(BigInteger, ForeignKey('cases.id'), nullable=False, comment='案件ID')
    field_id = Column(BigInteger, ForeignKey('standard_fields.id'), nullable=False, comment='字段ID')
    field_value = Column(String(2000), comment='字段值（JSON格式存储复杂类型）')
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # 关系
    case = relationship("Case", back_populates="standard_field_values")


class CaseCustomFieldValue(Base):
    """自定义字段值表"""
    __tablename__ = "case_custom_field_values"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    case_id = Column(BigInteger, ForeignKey('cases.id'), nullable=False, comment='案件ID')
    field_id = Column(BigInteger, ForeignKey('custom_fields.id'), nullable=False, comment='自定义字段ID')
    field_value = Column(String(2000), comment='字段值（JSON格式存储复杂类型）')
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # 关系
    case = relationship("Case", back_populates="custom_field_values")

File: backend-java/src/main/java/com/cco/model/entity/Case.java
package com.cco.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 案件主表
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("cases")
public class Case extends BaseEntity {

    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 案件唯一标识
     */
    private String caseCode;

    /**
     * 所属甲方ID
     */
    private Long tenantId;

    /**
     * 所属催收机构ID
     */
    private Long agencyId;

    /**
     * 所属催收小组ID
     */
    private Long teamId;

    /**
     * 分配催员ID
     */
    private Long collectorId;

    /**
     * 所属队列ID
     */
    private Long queueId;

    /**
     * 用户编号
     */
    private String userId;

    /**
     * 用户姓名
     */
    private String userName;

    /**
     * 手机号
     */
    private String mobile;

    /**
     * 案件状态：pending_repayment/partial_repayment/normal_settlement/extension_settlement
     */
    private String caseStatus;

    /**
     * 产品名称
     */
    private String productName;

    /**
     * App名称
     */
    private String appName;

    /**
     * 逾期天数（用于自动分配队列）
     */
    private Integer overdueDays;

    /**
     * 贷款金额
     */
    private BigDecimal loanAmount;

    /**
     * 已还款金额
     */
    private BigDecimal repaidAmount;

    /**
     * 逾期金额
     */
    private BigDecimal outstandingAmount;

    /**
     * 到期日期
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dueDate;

    /**
     * 结清日期
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime settlementDate;

    /**
     * 分配时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime assignedAt;

    /**
     * 最后联系时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime lastContactAt;

    /**
     * 下次跟进时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime nextFollowUpAt;

    /**
     * 是否停留（独立状态字段，与case_status分离）
     */
    private Boolean isStay;

    /**
     * 停留时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime stayAt;

    /**
     * 停留操作人ID
     */
    private Long stayBy;

    /**
     * 解放停留时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime stayReleasedAt;

    /**
     * 解放停留操作人ID
     */
    private Long stayReleasedBy;

}

